TACC-Cloud/geoapi
# task/DES-1828 Streetview

## Overview: ##

## PR Status: ##


## Related Jira tickets: ##

* [DES-1839](https://jira.tacc.utexas.edu/browse/DES-1839)
* [DES-1828](https://jira.tacc.utexas.edu/browse/DES-1828)
* [DES-1893](https://jira.tacc.utexas.edu/browse/DES-1893)

## Summary of Changes: ##
  * [X] Authenticate to Mapillary
  * [X] Authenticate to Google (Not needed right now)
* [X] Create a progress notification to poll upload results.
  * [X] Implement retry logic 
  * [X] Create a separate model to include progress.
  * [X] Aggregate progress messages in the log.
* [X] Handle upload from tapis to geoapi.
  * [X] Implement retry logic
  * [X] Ensure images contain supported file data (e.g. GPano and Exif)
* [X] Handle upload from geoapi to tapis.
  * [X] Parse stdout of subprocess to get progress string. 
  * [X] Handle error from progress string.
* [X] Separate model to link mapillary sequences and tapis systems and paths.
* [X] Handle sequence data from mapillary. 
* [X] Pre/Post-processing for mapillary
* [X] Expose more Mapillary APIs (e.g. changeset)
* [X] Implement Google Streetview Upload
* [X] ~~Abort mid-upload~~ There isn't a good way to undo once uploaded to mapillary
* [X] ~~Handle pending approval~~ Done in Hazmapper


## Testing Steps: ##
 - Run tests
 - Make sure it works with TACC-Cloud/hazmapper#45

## Notes: ##
